src/Runner.Listener/JobDispatcher.cs

```diff
@@ -767,6 +771,25 @@ public async Task RenewJobRequestAsync(int poolId, long requestId, Guid lockToke
             }
         }
 
+        private void UpdateAgentNameIfNeeded(string agentName)
+        {
+            try
+            {
+                if (_runnerSettings.AgentName != agentName)
```

Comment:
prefer `String.Equals` here, we can provide the default StringComparison.Ordinal just to be more explicit